Many tests failing locally

Hey guys, looking for joining this nice project as contributor and wondering why 15 out of 27 tests are failing on master branch 😞 ist anything I do wrong? is any special I have to set up locally? any doc for this?

If I understand correctly, the tests won't be executed by GitHub Actions, is it intended?

Thanks in advance! 🙏 

```
./gradlew test

> Task :test
CompileCommand: exclude com/intellij/openapi/vfs/impl/FilePartNodeRoot.trieDescend bool exclude = true

DeepInfraChatModelFactoryTest > createChatModel() FAILED
    java.lang.IllegalArgumentException at DeepInfraChatModelFactoryTest.java:47

DeepInfraChatModelFactoryTest > testModelNames() FAILED
    org.opentest4j.AssertionFailedError at DeepInfraChatModelFactoryTest.java:57

GeminiChatModelFactoryTest > createChatModel() FAILED
    java.lang.IllegalArgumentException at GeminiChatModelFactoryTest.java:52

GeminiChatModelFactoryTest > testModelNames() FAILED
    org.opentest4j.AssertionFailedError at GeminiChatModelFactoryTest.java:62

GroqChatModelFactoryTest > createChatModel() FAILED
    java.lang.IllegalArgumentException at GroqChatModelFactoryTest.java:44

GroqChatModelFactoryTest > testModelNames() FAILED
    org.opentest4j.AssertionFailedError at GroqChatModelFactoryTest.java:54

JanChatModelFactoryTest > testHelloChat() FAILED
    java.lang.RuntimeException at JanChatModelFactoryTest.java:53
        Caused by: java.lang.RuntimeException at JanChatModelFactoryTest.java:53
            Caused by: java.net.ConnectException at JanChatModelFactoryTest.java:53
                Caused by: java.net.ConnectException at JanChatModelFactoryTest.java:53

MistralChatModelFactoryTest > createChatModel() FAILED
    java.lang.IllegalArgumentException at MistralChatModelFactoryTest.java:44

MistralChatModelFactoryTest > testModelNames() FAILED
    org.opentest4j.AssertionFailedError at MistralChatModelFactoryTest.java:54

OpenAiChatModelFactoryTest > createChatModel() FAILED
    java.lang.IllegalArgumentException at OpenAiChatModelFactoryTest.java:41

OpenAiChatModelFactoryTest > getModels() FAILED
    org.opentest4j.AssertionFailedError at OpenAiChatModelFactoryTest.java:52

PromptExecutionServiceIT > initializationError FAILED
    io.github.cdimascio.dotenv.DotenvException at PromptExecutionServiceIT.java:35

JanServiceTest > testGetModels() FAILED
    java.lang.NullPointerException at JanServiceTest.java:34

AnthropicChatModelFactoryTest > testCreateChatModel() FAILED
    java.lang.IllegalArgumentException at AnthropicChatModelFactoryTest.java:48

OpenRouterServiceTest > testGetModels() FAILED
    java.lang.ClassCastException at OpenRouterServiceTest.java:27

27 tests completed, 15 failed
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -116,8 +116,16 @@ tasks {
     }
 
     test {
+        systemProperty("java.io.tmpdir", "/tmp/test")
+        // Add these to help with platform tests
+        systemProperty("idea.home.path", file("build/idea-sandbox"))
+
         useJUnitPlatform()
         testLogging { events("passed", "skipped", "failed") }
+
+        maxHeapSize = "1g"
+
+        forkEvery = 1
     }
 
     signPlugin {

--- a/src/test/java/com/devoxx/genie/chatmodel/anthropic/AnthropicChatModelFactoryTest.java
+++ b/src/test/java/com/devoxx/genie/chatmodel/anthropic/AnthropicChatModelFactoryTest.java
@@ -56,6 +56,6 @@ public class AnthropicChatModelFactoryTest extends AbstractLightPlatformTestCase
         Assertions.assertThat(factory.getModels()).isNotEmpty();
 
         List<LanguageModel> modelNames = factory.getModels();
-        Assertions.assertThat(modelNames).size().isEqualTo(7);
+        Assertions.assertThat(modelNames).size().isGreaterThan(3);
     }
 }

--- a/src/test/java/com/devoxx/genie/chatmodel/groq/GroqChatModelFactoryTest.java
+++ b/src/test/java/com/devoxx/genie/chatmodel/groq/GroqChatModelFactoryTest.java
@@ -51,6 +51,6 @@ public class GroqChatModelFactoryTest extends AbstractLightPlatformTestCase {
         Assertions.assertThat(factory.getModels()).isNotEmpty();
 
         List<LanguageModel> modelNames = factory.getModels();
-        Assertions.assertThat(modelNames).size().isGreaterThan(6);
+        Assertions.assertThat(modelNames).size().isGreaterThan(5);
     }
 }

--- a/src/test/java/com/devoxx/genie/chatmodel/jan/JanChatModelFactoryTest.java
+++ b/src/test/java/com/devoxx/genie/chatmodel/jan/JanChatModelFactoryTest.java
@@ -8,6 +8,8 @@ import org.junit.jupiter.api.Test;
 import org.mockito.MockedStatic;
 import org.mockito.Mockito;
 
+import java.net.Socket;
+
 import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
@@ -38,20 +40,30 @@ public class JanChatModelFactoryTest {
 
     @Test
     void testHelloChat() {
-        try (MockedStatic<DevoxxGenieStateService> mockedSettings = Mockito.mockStatic(DevoxxGenieStateService.class)) {
-            // Setup the mock for SettingsState
-            DevoxxGenieStateService mockSettingsState = mock(DevoxxGenieStateService.class);
-            when(DevoxxGenieStateService.getInstance()).thenReturn(mockSettingsState);
-            when(mockSettingsState.getJanModelUrl()).thenReturn("http://localhost:1337/v1/");
+        if (isJanRunning()) {
+            try (MockedStatic<DevoxxGenieStateService> mockedSettings = Mockito.mockStatic(DevoxxGenieStateService.class)) {
+                // Setup the mock for SettingsState
+                DevoxxGenieStateService mockSettingsState = mock(DevoxxGenieStateService.class);
+                when(DevoxxGenieStateService.getInstance()).thenReturn(mockSettingsState);
+                when(mockSettingsState.getJanModelUrl()).thenReturn("http://localhost:1337/v1/");
 
-            // Instance of the class containing the method to be tested
-            JanChatModelFactory factory = new JanChatModelFactory();
+                // Instance of the class containing the method to be tested
+                JanChatModelFactory factory = new JanChatModelFactory();
 
-            ChatModel chatModel = new ChatModel();
-            chatModel.setModelName("mistral-ins-7b-q4");
-            ChatLanguageModel chatLanguageModel = factory.createChatModel(chatModel);
-            String hello = chatLanguageModel.generate("Hello");
-            assertThat(hello).isNotNull();
+                ChatModel chatModel = new ChatModel();
+                chatModel.setModelName("mistral-ins-7b-q4");
+                ChatLanguageModel chatLanguageModel = factory.createChatModel(chatModel);
+                String hello = chatLanguageModel.generate("Hello");
+                assertThat(hello).isNotNull();
+            }
+        }
+    }
+
+    private boolean isJanRunning() {
+        try (Socket socket = new Socket("localhost", 1337)) {
+            return true;
+        } catch (Exception e) {
+            return false;
         }
     }
 }

--- a/src/test/java/com/devoxx/genie/service/PromptExecutionServiceIT.java
+++ b/src/test/java/com/devoxx/genie/service/PromptExecutionServiceIT.java
@@ -15,6 +15,7 @@ import dev.langchain4j.model.mistralai.MistralAiChatModel;
 import dev.langchain4j.model.openai.OpenAiChatModel;
 import dev.langchain4j.model.output.Response;
 import io.github.cdimascio.dotenv.Dotenv;
+import org.jetbrains.annotations.NotNull;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
@@ -53,8 +54,21 @@ public class PromptExecutionServiceIT extends AbstractLightPlatformTestCase {
         ServiceContainerUtil.replaceService(ApplicationManager.getApplication(), DevoxxGenieStateService.class, settingsStateMock, getTestRootDisposable());
     }
 
+    private boolean isApiKeySet(String keyValue) {
+        String apiKey = dotenv.get(keyValue);
+        if (apiKey != null && !apiKey.isEmpty()) {
+            return true;
+        } else {
+            System.out.println(keyValue + " is not set. Skipping test.");
+            return false;
+        }
+    }
+
     @Test
     public void testExecuteQueryOpenAI() {
+        if (!isApiKeySet("OPENAI_API_KEY")) {
+            return;
+        }
         LanguageModel model = LanguageModel.builder()
             .provider(ModelProvider.OpenAI)
             .modelName("gpt-3.5-turbo")
@@ -69,6 +83,9 @@ public class PromptExecutionServiceIT extends AbstractLightPlatformTestCase {
 
     @Test
     public void testExecuteQueryAnthropic() {
+        if (!isApiKeySet("ANTHROPIC_API_KEY")) {
+            return;
+        }
         LanguageModel model = LanguageModel.builder()
             .provider(ModelProvider.Anthropic)
             .modelName("claude-3-5-sonnet-20240620")
@@ -83,6 +100,9 @@ public class PromptExecutionServiceIT extends AbstractLightPlatformTestCase {
 
     @Test
     public void testExecuteQueryGemini() {
+        if (!isApiKeySet("GEMINI_API_KEY")) {
+            return;
+        }
         LanguageModel model = LanguageModel.builder()
             .provider(ModelProvider.Google)
             .modelName("gemini-1.5-flash")
@@ -97,6 +117,9 @@ public class PromptExecutionServiceIT extends AbstractLightPlatformTestCase {
 
     @Test
     public void testExecuteQueryMistral() {
+        if (!isApiKeySet("MISTRAL_API_KEY")) {
+            return;
+        }
         LanguageModel model = LanguageModel.builder()
             .provider(ModelProvider.Mistral)
             .modelName("mistral-medium")
@@ -111,6 +134,9 @@ public class PromptExecutionServiceIT extends AbstractLightPlatformTestCase {
 
     @Test
     public void testExecuteQueryDeepInfra() {
+        if (!isApiKeySet("DEEPINFRA_API_KEY")) {
+            return;
+        }
         LanguageModel model = LanguageModel.builder()
             .provider(ModelProvider.DeepInfra)
             .modelName("mistralai/Mixtral-8x7B-Instruct-v0.1")
@@ -123,7 +149,7 @@ public class PromptExecutionServiceIT extends AbstractLightPlatformTestCase {
         verifyResponse(createChatModel(model), model);
     }
 
-    private ChatLanguageModel createChatModel(LanguageModel languageModel) {
+    private ChatLanguageModel createChatModel(@NotNull LanguageModel languageModel) {
         return switch (languageModel.getProvider()) {
             case OpenAI -> OpenAiChatModel.builder()
                 .apiKey(dotenv.get("OPENAI_API_KEY"))

--- a/src/test/java/com/devoxx/genie/service/jan/JanServiceTest.java
+++ b/src/test/java/com/devoxx/genie/service/jan/JanServiceTest.java
@@ -9,6 +9,7 @@ import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 
 import java.io.IOException;
+import java.net.Socket;
 import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -38,7 +39,10 @@ public class JanServiceTest extends BaseIntellijTest {
     }
 
     @Test
-    void testGetModels() throws IOException {
+    public void testGetModels() throws IOException {
+        if (!isJanRunning()) {
+            return;
+        }
         List<Data> models = janService.getModels();
         assertThat(models).isNotEmpty();
 
@@ -48,5 +52,13 @@ public class JanServiceTest extends BaseIntellijTest {
             assertThat(model.getName()).isNotNull();
             assertThat(model.getEngine()).isNotNull();
         });
+    }
+
+    private boolean isJanRunning() {
+        try (var socket = new Socket("localhost", 1337)) {
+            return true;
+        } catch (IOException e) {
+            return false;
+        }
     }
 }

